### ISSUE TITLE
Pagination: Links prev and next don't need to be always links

### ISSUE THREAD

**Issue by marcoscv-work** [liferay/clay#5297]
In cases Next page or Prev page don't have page to go they must not be links.

Example, when we have only a page, we cannot go next or prev page because there are not pages to go

At the same time I see some strange tabindex="0" on `span` in links that make them by mandatory get the focus, even when we don't use Screen reader.
<img width="1097" alt="image" src="https://user-images.githubusercontent.com/7963804/212676382-84e3495a-037d-459d-a39f-73db4f194d7f.png">

**Hot to fix / recommendation**:

- Remove links when "Next page" or "Prev page" don't have page to go they must not be links.
- Remove exatra tabindex in the link

**Based on DXP issues like this**:
https://issues.liferay.com/browse/LPS-168538
